src/apps/chifra/pkg/rpcClient/get_receipt.go

```diff
@@ -123,17 +123,20 @@ func GetTransactionReceipt(chain string, bn uint64, txid uint64, txHash *common.
 
 // getRawTransactionReceipt fetches raw transaction. If txHash is provided, the function
 // will not fetch the transaction (we may have already loaded it)
-func getRawTransactionReceipt(chain string, bn uint64, txid uint64, txHash *common.Hash) (receipt *types.RawReceipt, tx ethTypes.Transaction, err error) {
+func getRawTransactionReceipt(chain string, bn uint64, txid uint64, txHash *common.Hash) (receipt *types.RawReceipt, tx *ethTypes.Transaction, err error) {
 	var txHashString string
-	if txHash != nil {
-		txHashString = txHash.Hex()
-	} else {
-		tx, err = TxFromNumberAndId(chain, bn, txid)
-		if err != nil {
-			return
-		}
-		txHashString = tx.Hash().Hex()
+	// TODO: this optimization makes it hard to know what this function will return.
+	// TODO: A code expecting tx can still supply txHash, in which case it will get nil.
+	// if txHash != nil {
```

Comment:
Can you remove these comments?

Reply:
Sure, but I think this is an important issue. Whole `rpcClient` package doesn't support the idea of "I already have some parts of the data", as far as I know. Maybe we could add an issue.